File: typescript/src/n0020_valid_parentheses.test.ts
import { expect } from "jsr:@std/expect";
import { valid_parentheses } from "./n0020_valid_parentheses.ts";

Deno.test("isValid", () => {
  expect(valid_parentheses("()")).toBe(true);
  expect(valid_parentheses("()[]{}")).toBe(true);
  expect(valid_parentheses("(]")).toBe(false);
  expect(valid_parentheses("([])")).toBe(true);
});

File: rust/src/n0020_valid_parentheses.test.rs
use crate::n0020_valid_parentheses::*;

#[test]
fn test_valid_parentheses() {
    assert!(valid_parentheses("()"));
    assert!(valid_parentheses("()[]{}"));
    assert!(!valid_parentheses("(]"));
    assert!(valid_parentheses("([])"));
}
